### PR TITLE
kie-issue#206: Enable autocomplete suggestions for Literal Expressions and Decision Table input cells on DMN Editor's Boxed Expression Editor

### LINKED ISSUE #206: Enable autocomplete suggestions for Literal Expressions and Decision Table input cells on DMN Editor's Boxed Expression Editor
```[tasklist]
### Subtasks
- [ ] FEEL Autocompletions should offer Input Data nodes names
```

```[tasklist]
### Acceptance criteria
- [ ] Write a nice description for this issue.
- [ ] Add acceptance criteria here.
```

### PR DESCRIPTION
**Pull request by danielzhe**
Closes: https://github.com/apache/incubator-kie-issues/issues/206

Enables autocomplete for FEEL expressions.

https://github.com/kiegroup/kie-tools/assets/7305741/02ed5222-19a0-4ea1-9d72-2a6a8ef4482c

### CODE REVIEW

**Review by jomarko — COMMENTED** [review]
Thank you for this cool feature @danielzhe . I found some interesting scenario when monaco starts to offer fields of nested structures, I think it offered me inappropriate options on the second level. See :
Screenshot 2023-10-20 131325

and try to model same decision in: 
[types.dmn.txt](https://github.com/kiegroup/kie-tools/files/13053609/types.dmn.txt)

Also, am I correct imported models are not partof this PR, right? https://github.com/apache/incubator-kie-issues/issues/539

**Comment by danielzhe** [comment]
Thank you, @jomarko !
That options that Monaco suggest by itself when it doesn't have any valid options available, so it tries to "discover" some options that "maybe is ok".  I don't know if we can disable this feature.

About the included models, they are not part of this PR yet. It's gonna take a few more time to make that feature available because we have to "parse" all the imported nodes and we don't want to load all models using the old Marshaller in GWT side because of performance reasons and it can lead to a lot of wasted work as soon as we finish the migration to a full non-GWT editor. @tiagobento is working on that.

**Comment by tiagobento** [comment]
@danielzhe There is a way to disable these suggestions from Monaco. I did it for SWF, but I don't remember how :D

**Comment by danielzhe** [comment]
@tiagobento Fixed the issue about recursive calls that we talked in pvt.

**Comment by danielzhe** [comment]
@jomarko @tiagobento Changes applied, ready for your review. I also did an improvement to allow space to trigger the autocomplete suggestions.

File: packages/feel-input-component/src/FeelConfigs.ts [hunk 1]
```diff
@@ -83,6 +83,7 @@ export const feelDefaultConfig = (
     minimap: {
       enabled: false,
     },
+    wordBasedSuggestions: false,
```
Comment (tiagobento) [inline packages/feel-input-component/src/FeelConfigs.ts:86]: That's the one!

**Review by tiagobento — APPROVED** [review]

**Review by yesamer — APPROVED** [review]
LGTM, thank you @danielzhe for this amazing feature. 
Please consider implementing a Test Suite for this new feature.

**Review by jomarko — APPROVED** [review]
I can confirm, in situation described originally as https://github.com/kiegroup/kie-tools/pull/2018#pullrequestreview-1689835344 are not these 'wrong' autosuggestions present anymore.

In the same scenario, currently no suggestions are present, what is probably expected? At least that is my feelingreading comments from @danielzhe and @tiagobento 

Screenshot 2023-10-25 141819


Thank you for the PR!

**Pull request merged**